src/javascript/app/pages/trade/charts/highchart.js

```diff
@@ -608,12 +608,14 @@ const Highchart = (() => {
                     i++;
                 }
                 const last = parseInt(last_data.x || last_data[0]);
-                if (last > (end_time * 1000) || last > ((sell_time || sell_spot_time) * 1000)) {
+                if (last >= (end_time * 1000) || last >= ((sell_time || sell_spot_time) * 1000)) {
                     stop_streaming = true;
                 } else {
                     // add a null point if the last tick is before end time to bring end time line into view
                     const time = isSoldBeforeExpiry() ? (sell_time || sell_spot_time) : end_time;
-                    chart.series[0].addPoint({ x: ((time || window.time.unix()) + margin) * 1000, y: null });
+                    const point = { x: time * 1000, y: null };
+                    chart.addSeries({ data: [point] });
+                    chart.zoomOut(); // sometimes navigator doesn't show new points
```

Comment:
if we are zooming out instead of adding a null point with the margin value, maybe we can update the above comment